src/main/java/ch/njol/skript/util/Utils.java

```diff
@@ -714,5 +711,19 @@ public static long parseLong(final String s) {
 		}
 		return lastIndex;
 	}
+
+	// Removes the first n elements from a Collection
+	public <E> boolean removeFirstIf(Collection<E> coll, int n, Predicate<E> filter) {
+		Iterator<E> each = coll.iterator();
+		boolean removed = false;
+
+		while (each.hasNext() && n-- > 0) {
```

Comment:
i'm thinking this might be more readable as a for loop: `for (; each.hasNext() && n > 0; n--)`

Reply:
This was originally made by Frank but I didn't use it cause I didn't get it much and kept what I understood, should I ask him to improve it or remove it and let him make it later?